src/api/app/models/package.rb

```diff
@@ -308,12 +308,14 @@ def is_locked?
     project.is_locked?
   end
 
+  def master_product_object
+    # test _product permissions if any other _product: subcontainer is used and _product exists
+    product_object = project.packages.find_by(name: "_product") if name =~ /\A_product:\w[-+\w\.]*\z/
```

Comment:
Couldn't we set the "master_product_object" when the '_product:*' package object get's created? Like an association